src/isomorphic/hooks/ReactComponentTreeHook.js

```diff
@@ -324,13 +339,27 @@ var ReactComponentTreeHook = {
     }
 
     var currentOwner = ReactCurrentOwner.current;
-    if (currentOwner && typeof currentOwner._debugID === 'number') {
-      var id = currentOwner && currentOwner._debugID;
-      info += ReactComponentTreeHook.getStackAddendumByID(id);
+    if (currentOwner) {
+      if (typeof currentOwner.tag === 'number') {
+        const fiber = ((currentOwner : any) : Fiber);
+        info += ReactComponentTreeHook.getStackAddendumByFiber(fiber);
+      } else if (typeof currentOwner._debugID === 'number') {
+        info += ReactComponentTreeHook.getStackAddendumByID(currentOwner._debugID);
+      }
     }
     return info;
   },
 
+  getStackAddendumByFiber(fiber : Fiber) : string {
+    var info = '';
+    var node = fiber;
+    do {
+      info += describeFiber(node);
+      node = node.return;
```

Comment:
It seems like this `.return` can be wrong (can be the parent's alternate rather than the parent). The type will never differ and the owner rarely will, but the source could. You can imagine something like this:

```
if (!isFancy) {
  return <div>{props.children}</div>;
}
return <div className={...} ref={...} style={...}>{props.children}</div>;
```

In that case, it would be best if we could show the correct source. Perhaps you could use ReactFiberTreeReflection to find the correct one. If you're not going to tackle this now, can you at least add a TODO?